### PR TITLE
Add support for `include()` in `MODULE.bazel`

### LINKED ISSUE #1804: Gazelle should extract module mappings from imported foo.MODULE.bazel files
### What version of gazelle are you using?

0.36.0

### What version of rules_go are you using?

0.47.1

### What version of Bazel are you using?

7.2.0rc1

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Linux/AMD64

### What did you do?

Moved rules_go import/setup to separate go.MODULE.bazel and imported from MODULE.bazel

### What did you expect to see?

No changes to rules_go imports

### What did you see instead?

When running Gazelle, imports from rules_go change from our apparent name to `@@rules_go`, e.g.:

```
+        "@@rules_go~//go/runfiles",
         "@com_github_stretchr_testify//suite",
-        "@io_bazel_rules_go//go/runfiles:go_default_library",
```

### PR DESCRIPTION
**Pull request by fmeum**
**What type of PR is this?**

Bug fix

**What package or component does this PR mostly affect?**

cmd/gazelle

**What does this PR do? Why is it needed?**

Previously, `gazelle` wouldn't look into `include`ed segments of module files when collecting the apparent to module name mapping.

**Which issues(s) does this PR fix?**

Fixes #1804

**Other notes for review**

### CODE REVIEW

**Comment by fmeum** [comment]
@mortenmj Could you test this?

**Comment by tyler-french** [comment]
Tested at Uber with Bazel 7.1.1

**Review by tyler-french — APPROVED** [review]

File: internal/module/module.go [hunk 1]
```diff
@@ -61,11 +53,60 @@ func parseModuleFile(repoRoot string) (*build.File, error) {
...
+		names, includeLabels := collectApparentNamesAndIncludes(bf)
+		for name, apparentName := range names {
+			apparentNames[name] = apparentName
+		}
+		for _, includeLabel := range includeLabels {
+			var l label.Label
+			l, err = label.Parse(includeLabel)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse include label %q: %v", includeLabel, err)
+			}
```
Comment (tyler-french) [inline internal/module/module.go:86]: no need to declare the label
```suggestion
			l, err := label.Parse(includeLabel)
			if err != nil {
				return nil, fmt.Errorf("failed to parse include label %q: %v", includeLabel, err)
			}
```

File: internal/module/module.go [hunk 2]
```diff
@@ -61,11 +53,60 @@ func parseModuleFile(repoRoot string) (*build.File, error) {
 // Collects the mapping of module names (e.g. "rules_go") to user-configured apparent names (e.g.
 // "my_rules_go"). See https://bazel.build/external/module#repository_names_and_strict_deps for more
 // information on apparent names.
-func collectApparentNames(m *build.File) map[string]string {
+func collectApparentNames(repoRoot, relPath string) (map[string]string, error) {
```
Comment (tyler-french) [inline internal/module/module.go:56]: We should probably add a test for this function

Comment (fmeum) [reply internal/module/module.go:56]: I added a test and it also found a bug - I didn't know that `Rule` has an `ImplicitName`.

**Pull request merged**